js-sdk-framework-tests/nextjs/src/lib/app_tests/ai/test.ts

```diff
@@ -0,0 +1,156 @@
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { initializeApp } from 'firebase/app';
+import {
+  GenerationConfig,
+  HarmBlockThreshold,
+  HarmCategory,
+  Content,
+  SafetySetting,
+  getAI,
+  getGenerativeModel,
+  GoogleAIBackend
+} from 'firebase/ai';
+import { getAuth, signInAnonymously } from 'firebase/auth';
+import { firebaseConfig } from '@/lib/app_tests/firebase';
+import { OK, FAILED } from '@/lib/app_tests/util';
+
+export type TestResults = {
+  initializeAppResult: string,
+  initializeAuthResult: string,
+  signInAnonymouslyResult: string,
+  getAIResult: string,
+  getGenerativeModelResult: string,
+  startChatResult: string,
+  chatSendFirstMessageResult: string,
+  chatFirstResponseCheckResult: string,
+  chatSendSecondMessageResult: string,
+  chatSecondResponseCheckResult: string,
+  getHistoryResult: string,
+  deleteUserResult: string
+};
+
+export function initializeTestResults(): TestResults {
+  const testAnalyticsResult: TestResults = {
+    initializeAppResult: FAILED,
+    initializeAuthResult: FAILED,
+    signInAnonymouslyResult: FAILED,
+    getAIResult: FAILED,
+    getGenerativeModelResult: FAILED,
+    startChatResult: FAILED,
+    chatSendFirstMessageResult: FAILED,
+    chatFirstResponseCheckResult: FAILED,
+    chatSendSecondMessageResult: FAILED,
+    chatSecondResponseCheckResult: FAILED,
+    getHistoryResult: FAILED,
+    deleteUserResult: FAILED
+  };
+  return testAnalyticsResult;
+}
+
+const commonGenerationConfig: GenerationConfig = {
+  temperature: 0,
+  topP: 0,
+  responseMimeType: 'text/plain'
+};
+
+const commonSafetySettings: SafetySetting[] = [
+  {
+    category: HarmCategory.HARM_CATEGORY_HARASSMENT,
+    threshold: HarmBlockThreshold.BLOCK_LOW_AND_ABOVE
+  },
+  {
+    category: HarmCategory.HARM_CATEGORY_HATE_SPEECH,
+    threshold: HarmBlockThreshold.BLOCK_LOW_AND_ABOVE
+  },
+  {
+    category: HarmCategory.HARM_CATEGORY_SEXUALLY_EXPLICIT,
+    threshold: HarmBlockThreshold.BLOCK_LOW_AND_ABOVE
+  },
+  {
+    category: HarmCategory.HARM_CATEGORY_DANGEROUS_CONTENT,
+    threshold: HarmBlockThreshold.BLOCK_LOW_AND_ABOVE
+  }
+];
+
+const commonSystemInstruction: Content = {
+  role: 'system',
+  parts: [
+    {
+      text: 'You are a friendly and helpful assistant.'
+    }
+  ]
+};
+
+export async function testAI(isServer: boolean = false): Promise<TestResults> {
+  if (isServer) {
+    console.log("Server side");
+  }
+  const result: TestResults = initializeTestResults();
+  const firebaseApp = initializeApp(firebaseConfig);
+  result.initializeAppResult = OK;
+
+  const auth = getAuth(firebaseApp);
```

Comment:
Why do we sign in with auth for this test?

Reply:
Huh. At first I didn't have a user signed in and at that time the AI call failed, so I added as user login asnd.. changed a few other things at the same time and the AI call worked. I guess the few other things are what made it work, not signing in a user.

Long story short: removed the Auth sign in stuff. Thanks!